Add HTTPS/certificate authentication support to file transfer service (#2474)
thin-edge/thin-edge.io: docs/src/references/tedge-file-transfer-service.md

All uploaded files are preserved until they are explicitly deleted with the DELETE API.
To avoid exhaustion of storage space on the thin-edge device,
users must be diligent to delete any stored files as soon as their purpose is served.

> ## HTTPS and authenticated access
> By default, the service is unauthenticated and does not support HTTPS connections.

Reviewer: ```suggestion
## HTTPS and authenticated access

By default, the service is unauthenticated and does not support HTTPS connections.
```

HTTPS can be enabled by setting `http.cert_path` and `http.key_path`.
If the certificates are configured, the agent will automatically host the service via HTTPS, and redirect any
HTTP requests to the equivalent HTTPS URL.
If HTTPS is enabled, the configured certificate should be installed in the OS trust store for any connected agents
in order for them to trust the connection to the mapper.

Once HTTPS is enabled for the file-transfer service, certificate-based authentication can also be enabled.

> The directory containing the certificates that the mapper will trust can be configured using `http.ca_path`,

Reviewer: ```suggestion
The directory containing the certificates that the agent will trust can be configured using `http.ca_path`,
```

> and the agent can be configured to use a trusted certificate using the `http.client.auth.cert_file` and `http.client.auth.key_file`

Reviewer: ```suggestion
and the mapper can be configured to use a trusted certificate using the `http.client.auth.cert_file` and `http.client.auth.key_file`
```

Author: I don't think that's true, but the "mapper" in the sentence above is incorrect (as you can probably tell, I copied and pasted it from the auth proxy docs since they're broadly the same, but forgot to change that bit).

Reviewer: Huh? There seems to be some misunderstanding about this setting then, as I see @didier-wenzek also suggesting a similar change. This is the **`client`** setting, right? Meant to be used by the clients of the agent: which are the mappers and other child devices.

Author: I've adjusted it to be in line with Didier's comment, as well as correcting the line above, which was also adding to my confusion as that was also wrong.

> and the agent can be configured to use a trusted certificate using the `http.client.auth.cert_file` and `http.client.auth.key_file`

Reviewer 2: ```suggestion
and the mapper as well as the child device agents have to be configured to use a trusted certificate using the `http.client.auth.cert_file` and `http.client.auth.key_file`
```
